simulator/func_sim/alu.h

```diff
@@ -236,6 +236,18 @@ struct ALU
 
     // Bit manipulations
     template<typename I> static void sbext( I* instr) { instr->v_dst[0] = 1U & ( instr->v_src[0] >> shamt_v_src2<typename I::RegisterUInt>( instr)); }
+    
+    template<typename I> static 
+    void max( I* instr)
+    { 
+        instr->v_dst[0] = std::max( narrow_cast<typename I::RegisterUInt>( instr->v_src[0]), narrow_cast<typename I::RegisterUInt>( instr->v_src[1]));;
```

Comment:
This builds fine, but returns incorrect answer. (0xffffffff > 0x11111111) What is wrong with it?

Reply:
Please try to localize and fix your mistake.